caffe2/perfkernels/embedding_lookup_idx.cc

```diff
@@ -28,14 +28,14 @@ static bool EmbeddingLookupGenericSlowIdx(
     const float* scale_bias, // optional scale & bias params for uint8 input
     bool normalize_by_lengths,
     OutType* out) {
-  int64_t current = 0;
+  int64_t current = offsets[0];
```

Comment:
With the suggested changes on the caller side we can just set current to 0 here and at the end we can compare current == index_size